rpc/blockchain.go

```diff
@@ -227,7 +229,9 @@ func (s *PublicBlockchainService) GetBlockByNumber(
 			}
 		}
 
-		s.blockCache.Add(uint64(blockNum), response)
+		if blockNum != rpc.PendingBlockNumber {
+			s.blockCache.Add(blk.NumberU64(), response)
```

Comment:
can not add the response into cache directly, because we have 3 version rpc, and the blockArgs also affect the result.

Reply:
Fix it, please check it again